base/base64.hpp

```diff
@@ -0,0 +1,48 @@
+﻿
+#pragma once
+
+#include <cstdint>
+
+#include "base/encoder.hpp"
+
+namespace principia {
+namespace base {
+namespace internal_base64 {
+
+// This function implements RFC 4648 section 5 (base64url).  The encoded text is
+// *not* padded.
+template<bool null_terminated>
+class Base64Encoder : public Encoder<char, null_terminated> {
+ public:
+  // Encodes |input| into |output|, which must be large enough to hold the
+  // encoded form.
+  inline void Encode(Array<std::uint8_t const> input,
+                     Array<char> output) override;
+
+  // Same as above but the storage is allocated by the callee.
+  inline UniqueArray<char> Encode(Array<std::uint8_t const> input) override;
+
+  // Length of the encoded form, in char16_t.
```

Comment:
Do not add comments that are redundant with the base class, let alone contradictory (`char16_t`?).